libsolidity/interface/StandardCompiler.cpp

```diff
@@ -236,7 +238,7 @@ Json::Value StandardCompiler::compileInternal(Json::Value const& _input)
 		return formatFatalError("JSONError", "Only \"Solidity\" is supported as a language.");
 
 	Json::Value const& sources = _input["sources"];
-	if (!sources)
+	if (sources.empty())
```

Comment:
Is this a bug fix?

Reply:
Nope, they just removed `operator!()` in version `1.8.4`
```cpp
bool Value::operator!() const { return isNull(); }
```
Lol, now I was able to fully understand why I had problems using that operator. On my system where was somehow an old `jsoncpp` header used - will check the build system, how that was possible - and because they don't provided an implementation of `operator!()` I had linking problems. I just realised that they remove the `operator!()`, thats why I used `sources.empty()` here. Now I see that they also providing
```cpp
Value::operator bool() const { return ! isNull(); }
```
So everything is still fine. Wow, sorry for that!  The original code still works very nice with the new version `1.8.4`. I will remove that change.

Reply:
... turns out it was a bug fix
```cpp
bool Value::empty() const {
  if (isNull() || isArray() || isObject())
    return size() == 0u;
  else
    return false;
}
```
In contrast to `Value::operator bool() `, `Value::empty()` is also checking whether an empty array, or an empty object was provided. I just added two new test-cases in `test/libsolidity/StandardCompiler.cpp`

```cpp
BOOST_AUTO_TEST_CASE(no_sources_empty_array)
{
  char const* input = R"(
	{
		"language": "Solidity",
		"sources": []
	}
	)";
  Json::Value result = compile(input);
  BOOST_CHECK(containsError(result, "JSONError", "No input sources specified."));
}

BOOST_AUTO_TEST_CASE(no_sources_empty_object)
{
	char const* input = R"(
	{
		"language": "Solidity",
		"sources": {}
	}
	)";
	Json::Value result = compile(input);
	BOOST_CHECK(containsError(result, "JSONError", "No input sources specified."));
}
```

Reply:
Can you create a new PR which adds the test and removes the use of the `!` operator (without upgrading jsoncpp)?

Reply:
Sure!